core/templates/domain/exploration/RecordedVoiceoversObjectFactory.ts

```diff
@@ -58,28 +55,29 @@ export class RecordedVoiceovers {
   }
 
   markAllVoiceoversAsNeedingUpdate(contentId: string): void {
-    var languageCodeToVoiceover = this.voiceoversMapping[contentId];
-    for (var languageCode in languageCodeToVoiceover) {
-      languageCodeToVoiceover[languageCode].markAsNeedingUpdate();
-    }
+    Object.values(this.voiceoversMapping[contentId])
+      .forEach(voiceover => voiceover.markAsNeedingUpdate());
   }
 
   getVoiceoverLanguageCodes(contentId: string): string[] {
     return Object.keys(this.voiceoversMapping[contentId]);
   }
 
   hasVoiceovers(contentId: string): boolean {
-    return this.getVoiceoverLanguageCodes(contentId).length > 0;
+    return (
+      this.voiceoversMapping[contentId] &&
+      this.getVoiceoverLanguageCodes(contentId).length > 0);
+  }
+
+  hasVoiceover(contentId: string, langCode: string): boolean {
+    return (
+      this.voiceoversMapping.hasOwnProperty(contentId) &&
+      this.voiceoversMapping[contentId].hasOwnProperty(langCode));
   }
 
   hasUnflaggedVoiceovers(contentId: string): boolean {
-    var languageCodeToVoiceover = this.voiceoversMapping[contentId];
-    for (var languageCode in languageCodeToVoiceover) {
-      if (!languageCodeToVoiceover[languageCode].needsUpdate) {
-        return true;
-      }
-    }
-    return false;
+    return Object.values(this.voiceoversMapping[contentId])
```

Comment:
If contentId property does not exist, it might throw an error here.

Reply:
Fixed.